kernel/section.ml

```diff
@@ -20,7 +20,9 @@ type section_entry =
 
 type 'a entry_map = 'a Cmap.t * 'a Mindmap.t
 
-type 'a section = {
+type 'a t = {
+  sec_prev : 'a t option;
```

Comment:
Really not pretty, I'd prefer a list and ensuring everywhere that it's not empty.

Reply:
We use non-empty lists in a few places already so indeed we could add a generic `NEList` data structure to the libraries.

Reply:
TBH I don't see a need to use nonempty lists here, having the outer section be a member of the current one seems to work fine.